pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -286,4 +288,13 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.PodLogsDir == "" {
 		obj.PodLogsDir = DefaultPodLogsDir
 	}
+
+	if obj.SingleProcessOOMKill == nil && runtime.GOOS == "linux" {
+		if !kubeletutil.IsCgroup2UnifiedMode() {
+			// This is a default behavior for cgroups v1.
+			obj.SingleProcessOOMKill = utilpointer.Bool(true)
+		} else {
+			obj.SingleProcessOOMKill = utilpointer.Bool(false)
```

Comment:
why not document what the field being unset means in cgroupv1 mode and in cgroupv2 mode? serializing an API-level default into the config file based on the current machine the defaulting happens to run on makes it really hard to generate configs centrally using something like kubeadm for distribution